packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -372,30 +380,57 @@ class ClearSelectionEvent extends SelectionEvent {
   const ClearSelectionEvent(): super._(SelectionEventType.clear);
 }
 
+/// An abstract base class for selection events that select a boundary at a location.
+///
+/// See also:
+/// * [SelectWordSelectionEvent], for an event that selects the word at the location.
+/// * [SelectParagraphSelectionEvent], for an event that selects the paragraph at the location.
+/// * [SelectLineSelectionEvent], for an event that selects the line at the location.
+/// * [SelectionEventType], for determining the subclass types.
+abstract class SelectBoundarySelectionEvent extends SelectionEvent {
+  const SelectBoundarySelectionEvent._(this.globalPosition, SelectionEventType type) : super._(type);
+
+  /// The position in global coordinates to select the boundary at.
+  final Offset globalPosition;
+}
+
+/// An abstract base class for selection events that select a boundary that
+/// may span multiple [Selectable]s.
+///
+/// See also:
+/// * [SelectParagraphSelectionEvent], for an event that selects the paragraph at the location.
+/// * [SelectLineSelectionEvent], for an event that selects the line at the location.
+/// * [SelectionEventType], for determining the subclass types.
+abstract class SelectMultiSelectableBoundarySelectionEvent extends SelectBoundarySelectionEvent {
+  const SelectMultiSelectableBoundarySelectionEvent._(this.absorb, Offset globalPosition, SelectionEventType type) : super._(globalPosition, type);
+
+  /// Whether the selectable receiving the event should be absorbed into
+  /// an encompassing boundary.
```

Comment:
Can you explain a bit more in the doc? what is encompassing boundary?

Reply:
Added some more docs in [`bf5d04c` (#153514)](https://github.com/flutter/flutter/pull/153514/commits/bf5d04c50816a41e7a7b68c217f1bfcb96d4a1dc).